fractions/keycloak-microprofile-jwt/src/main/java/org/wildfly/swarm/keycloak/mpjwt/deployment/KeycloakJWTCallerPrincipalFactory.java

```diff
@@ -12,21 +13,37 @@
 import io.smallrye.jwt.auth.principal.JWTCallerPrincipal;
 import io.smallrye.jwt.auth.principal.JWTCallerPrincipalFactory;
 import io.smallrye.jwt.auth.principal.ParseException;
+
 /**
  * An implementation of the abstract JWTCallerPrincipalFactory that uses the Keycloak token parsing classes.
  */
 public class KeycloakJWTCallerPrincipalFactory extends JWTCallerPrincipalFactory {
+    private static final Logger log = Logger.getLogger(KeycloakJWTCallerPrincipalFactory.class);
+
+    private static final KeycloakDeployment deployment = createDeployment();
 
     @Override
     public JWTCallerPrincipal parse(final String token, final JWTAuthContextInfo authContextInfo) throws ParseException {
         try {
-            InputStream keycloakJson = Thread.currentThread().getContextClassLoader().getResourceAsStream("keycloak.json");
-            KeycloakDeployment deployment = KeycloakDeploymentBuilder.build(keycloakJson);
             JWSInput jwsInput = new JWSInput(token);
             AccessToken accessToken = AdapterTokenVerifier.verifyToken(jwsInput.getWireString(), deployment);
             return new KeycloakJWTCallerPrincipal(jwsInput.readContentAsString(), accessToken);
         } catch (Throwable ex) {
             throw new ParseException("Failure to parse the token", ex);
         }
     }
+
+    private static KeycloakDeployment createDeployment() {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        InputStream keycloakJson = cl.getResourceAsStream("keycloak.json");
+        if (keycloakJson == null) {
+            keycloakJson = cl.getResourceAsStream("WEB-INF/keycloak.json");
```

Comment:
Will this work? Not sure if `WEB-INF` is on "classpath"... Did you try both?

Reply:
I don't know, you suggested to add this check on IRC ? Or did you mean something else ?

Reply:
Hmm, looks like my response got lost...So, I don't know, you suggested add this check on IRC or did you mean something else ?

Reply:
Well I did suggest to look for `WEB-INF/keycloak.json`, as I know this is where `keycloak.json` is supposed to be put. I have no idea if this code will do that correctly :-)

Reply:
You are right the code I've added is ineffective.
OK, I've looked at [this code](https://github.com/thorntail/thorntail/blob/master/fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java).
So, the WF keycloak subsystem (=> undertow KC login code) does need keycloak.json in WEB-INF, but this fraction does not because it uses the adapter API directly.
That is not to say this fraction should not support it, but I would not mind looking at as an improvement request as opposed to a blocking issue for this PR. Or would you rather have it sorted out sooner rather than later ?

Reply:
I guess it makes sense to sort it out sooner rather than later for users to avoid copying keycloak.json if moving from 'keycloak' to this fraction

Reply:
Exactly. Also things outside of `WEB-INF` are publicly accessible, which we probably don't want here.

Reply:
I don't think it applies, [these keycloak.jsons](https://github.com/thorntail/thorntail-examples/tree/master/security/keycloak/src/main/resources), named differently in the demo, end up in WEB-INF/classes.